### PR TITLE
📝 Add Turn warnings into errors in Build documentation

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview
Add Turn warnings into errors in Build documentation

 

### Details

- This CI step will check that Sphinx documentation has no WARNING.

https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-W

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -245,7 +245,7 @@ jobs:
 
   # run this twice since VTK9 fails to close render windows
   - script: |
-      (make -C docs html -b coverage; make -C docs html -b coverage)
+      (make -C docs html -b coverage SPHINXOPTS="-W"; make -C docs html -b coverage)
```
Comment (tkoyama010) [inline azure-pipelines.yml:248]:
```suggestion
      (make -C docs html -b coverage; make -C docs html -b coverage SPHINXOPTS="-W")
```

File: azure-pipelines.yml [hunk 2]
```diff
@@ -245,7 +245,7 @@ jobs:
 
   # run this twice since VTK9 fails to close render windows
   - script: |
-      (make -C docs html -b coverage; make -C docs html -b coverage)
+      (make -C docs html -b coverage; make -C docs html -b coverage SPHINXOPTS="-W")
```
Comment (tkoyama010) [inline azure-pipelines.yml:248]:
```suggestion
      (make -C docs html -b coverage SPHINXOPTS="-W"; make -C docs html -b coverage)
```

File: docs/getting-started/simple.rst [hunk 3]
```diff
@@ -130,9 +130,6 @@ to each PyVista data object:
     mesh.plot(screenshot='airplane.png')
```
Comment (tkoyama010) [inline docs/getting-started/simple.rst:130]:
```suggestion
    mesh.plot(screenshot='./images/auto-generated/airplane.png')
```

File: docs/getting-started/simple.rst [hunk 4]
```diff
@@ -130,9 +130,6 @@ to each PyVista data object:
...
 
 
-.. image:: ../images/auto-generated/airplane.png
-
-
 You can also create the plotter to highly control the scene. First, instantiate
```
Comment (tkoyama010) [inline docs/getting-started/simple.rst:133]:
```suggestion
.. image:: ../images/auto-generated/airplane.png


You can also create the plotter to highly control the scene. First, instantiate
```

File: examples/02-plot/movie.py [hunk 5]
```diff
@@ -42,6 +42,3 @@
 
 # Be sure to close the plotter when finished
 plotter.close()
```
Comment (tkoyama010) [inline examples/02-plot/movie.py:44]:
```suggestion
plotter.close()

###############################################################################
# .. figure:: ../../images/auto-generated/sphere-shrinking.mp4
```

File: azure-pipelines.yml (hunk 1 above)
Comment (tkoyama010) [inline azure-pipelines.yml:248]:
```suggestion
      (make -C docs html -b coverage SPHINXOPTS="-W"; make -C docs html -b coverage SPHINXOPTS="-W")
```

File: examples/02-plot/movie.py [hunk 6]
```diff
@@ -9,7 +9,7 @@
 import pyvista as pv
 import numpy as np
 
-filename = "sphere-shrinking.mp4"
+filename = "./images/auto-generated/sphere-shrinking.mp4"
```
Comment (tkoyama010) [inline examples/02-plot/movie.py:12]:
```suggestion
filename = "sphere-shrinking.mp4"
```

File: examples/02-plot/movie.py [hunk 7]
```diff
@@ -42,6 +42,3 @@
 
 # Be sure to close the plotter when finished
 plotter.close()
-
-###############################################################################
-# .. figure:: ../../images/auto-generated/sphere-shrinking.mp4
```
Comment (tkoyama010) [inline examples/02-plot/movie.py:47]: This is not working.
https://dev.pyvista.org/examples/02-plot/movie.html
image

File: docs/getting-started/simple.rst [hunk 8]
```diff
@@ -127,10 +127,7 @@ to each PyVista data object:
     from pyvista import examples
 
     mesh = examples.load_airplane()
-    mesh.plot(screenshot='airplane.png')
-
-
-.. image:: ../images/auto-generated/airplane.png
```
Comment (tkoyama010) [inline docs/getting-started/simple.rst:133]: This part is not working.
https://dev.pyvista.org/getting-started/simple.html#plotting
image

**Review by tkoyama010 — COMMENTED** [review]
@pyvista/developers It is ready for review

File: docs/getting-started/simple.rst (hunk 8 above)
Comment (akaszynski) [reply docs/getting-started/simple.rst:133]: Fixed in https://github.com/pyvista/pyvista/pull/1238/commits/d0dca76fb98fd8fb6f35d0844a02259cd7a456dc.  Issue is our default backend changed to `ipyvtklink`, which is incompatible with sphinx jupyterlab.

File: examples/02-plot/movie.py (hunk 7 above)
Comment (akaszynski) [reply examples/02-plot/movie.py:47]: Just made a comment that the movie won't render on a webpage.

**Review by akaszynski — APPROVED** [review]
Thanks for going over the docs, and in favor of raising warnings as errors.

**Comment by tkoyama010** [comment]
@akaszynski Thank you for the appropriate changes. The majority of this change is your contribution.

**Comment by akaszynski** [comment]
> @akaszynski Thank you for the appropriate changes. The majority of this change is your contribution.

You pointed out the bugs in the docs, something we didn't see when we changed the default jupyter backend, which is probably more important (and time consuming) than the actual fix!

**Pull request merged**